components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/util/EncryptionUtil.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.keyrotation.util;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+import org.apache.axiom.om.util.Base64;
+import org.wso2.carbon.identity.keyrotation.config.KeyRotationConfig;
+import org.wso2.carbon.identity.keyrotation.model.CipherMetaData;
+import org.wso2.carbon.identity.keyrotation.service.CryptoProvider;
+
+import java.nio.charset.Charset;
+
+/**
+ * The class that calls the reEncryption mechanism.
+ */
+public class EncryptionUtil {
+
+
+    private static final Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+
+    /**
+     * ReEncryption mechanism needed for the key rotation task.
+     *
+     * @param cipher            The ciphertext needed to perform re-encryption on.
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @return Ciphertext that gets decrypted from the old key and encrypted using the new key.
+     * @throws KeyRotationException Exception thrown if something unexpected happens during key rotation.
+     */
+    public static String reEncryptor(String cipher, KeyRotationConfig keyRotationConfig)
```

Comment:
Shall we rename this function with an action without using a noun

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/13/commits/c1822aa3638f426a55388883199f2d5788aa91ae